### ISSUE TITLE
question about filters collectors

### ISSUE THREAD

**Issue by wkshare** [bosun-monitor/bosun#695]
if I input "scollector -f collectors.c_procstats_linux",  it can listen collectors.c_procstats_linux,which includes so many  items for me , I only need some of them. for example I only need  linux.loadavg. How can I do this ?

**Comment by kylebrandt** [comment]
The problem is that it is one giant function, that really should be refactored into a few smaller functions (that could still all go in the same file).

So for example in https://github.com/bosun-monitor/bosun/blob/master/cmd/scollector/collectors/procstats_linux.go we have

```
func init() {
	collectors = append(collectors, &IntervalCollector{F: c_procstats_linux})
}
```

And it should maybe be:

```
func init() {
	collectors = append(collectors, &IntervalCollector{F: c_procstats_linux_vmstat})
        collectors = append(collectors, &IntervalCollector{F: c_procstats_linux_stat})
        // ... etc
}
```

I don't have time to work on that anytime soon, but we can take a PR that changes this collector - I agree it is too much in one function.

**Comment by wkshare** [comment]
Thanks ！
How to reduce the sampling frequency？

**Comment by wkshare** [comment]
@kylebrandt

**Comment by kylebrandt** [comment]
Fixed in #697
